### PR TITLE
Checkout payment

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

## Other information

### CODE REVIEW

File: apps/foundation-demo/src/app/checkout/components/payment/payment-form/payment-form.component.scss [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    justify-self: center;
+    margin-bottom: 25px;
+    grid-area: note;
+    font-size: $small-font-size;
+  }
+
+  @include breakpoint(large) {
+    grid-template-columns: repeat(12, 1fr);
+    grid-template-areas: 
+      "        title         title         title         title         title         title         title   free-space1   free-space1   free-space1   free-space1   free-space1   free-space1   free-space1"
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply apps/foundation-demo/src/app/checkout/components/payment/payment-form/payment-form.component.scss]: nice

[the remark this answers is not in the document]
Comment (lderrickable) [reply apps/foundation-demo/src/app/checkout/components/payment/payment-form/payment-form.component.scss]: done. I changed the payment-form as well

**Comment by lderrickable** [comment]
@xelaint 
I updated the font-sizes for `shipping-summary` and `shipping-summary__title`, but the other two I couldn't do. I can't style the title of `shipping-options` through `shipping-summary` because `shipping-options` is its own component. I could put the radio buttons directly inside of `shipping-summary`, but I think `shipping-options` requires enough logic that it should be its own component.

I can't give `checkout__shipping` bottom-padding for really convoluted reasons, which basically come down to "I can't because the bottom element of `checkout__shipping` changes based on the state of the checkout app." The lengthy explanation is that there is margin that exists between `shipping-options` and the `continue-to-payment` button below it. When the `continue-to-payment` button is clicked, the button disappears, which makes `shipping-options`' bottom-margin the new bottom-margin of `checkout__shipping`. And that would mess up the `checkout__shipping` bottom-padding by making it appear bigger. BUT, as I'm typing this I thought of a possible way to fix this, which involves a discussion on how editing the shipping-option works. Maybe we can talk on Monday

File: apps/foundation-demo/src/app/checkout/actions/shipping.actions.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import { Action } from '@ngrx/store';
 
 export enum ShippingActionTypes {
-  ToggleShippingFormAction = "[Foundation-Shipping] Toggle Shipping Form Action",
+  ToggleShowShippingFormAction = "[Foundation-Shipping] Toggle Show Shipping Form Action",
```
Comment (damienwebdev) [inline apps/foundation-demo/src/app/checkout/actions/shipping.actions.ts]: This seems like it's outside the scope of this PR. Is it possible to cherry pick this commit into another PR?

File: apps/foundation-demo/src/app/checkout/components/shipping/shipping/shipping.component.html [hunk 3]
```diff
@@ -8,6 +8,8 @@
     *ngIf="shippingInfo"
```
Comment (damienwebdev) [inline apps/foundation-demo/src/app/checkout/components/shipping/shipping/shipping.component.html]: This API is starting to look absurd. Let's revisit this in a subsequent PR.

File: apps/foundation-demo/src/app/cart/components/cart-summary/cart-summary.component.html [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <div class="cart-summary" *ngIf="cart.items.length>0">
   <div class="cart-summary__title">cart summary</div>
-  <promotion></promotion>
+  <promotion class="cart-summary__promotion"></promotion>
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply apps/foundation-demo/src/app/cart/components/cart-summary/cart-summary.component.html]: I'm actually pulling this out of `cart-summary` and into `cart-async-wrapper` in a future PR. Is that ok? Or would you rather I did it here?

File: apps/foundation-demo/src/app/checkout/actions/payment.actions.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { Action } from '@ngrx/store';
+
+export enum PaymentActionTypes {
+  ShowPaymentViewAction = "[Foundation-Payment] Show Payment View Action",
```
[the remark this answers is not in the document]
Comment (lderrickable) [reply apps/foundation-demo/src/app/checkout/actions/payment.actions.ts]: done

File: apps/foundation-demo/src/app/checkout/actions/shipping.actions.ts (hunk 2 above)
Comment (lderrickable) [reply apps/foundation-demo/src/app/checkout/actions/shipping.actions.ts]: moved to WIP branch: https://github.com/graycoreio/daffodil/pull/39

File: apps/foundation-demo/src/app/cart/components/cart-summary/cart-summary.component.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (damienwebdev) [reply apps/foundation-demo/src/app/cart/components/cart-summary/cart-summary.component.html]: I think it's in the scope of this PR.

File: apps/foundation-demo/src/app/checkout/actions/payment.actions.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (damienwebdev) [reply apps/foundation-demo/src/app/checkout/actions/payment.actions.ts]: lgtm

File: apps/foundation-demo/src/app/cart/components/cart-summary/cart-summary.component.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (lderrickable) [reply apps/foundation-demo/src/app/cart/components/cart-summary/cart-summary.component.html]: done

**Comment by damienwebdev** [comment]
@lderrickable lgtm.
